djes/search.py

```diff
@@ -100,3 +100,73 @@ def execute(self):
                                                 **self._params))
 
         return self._executed
+
+
+class SearchParty(object):
+    """
+    Generator for multiple searches that allows for searches to iterate through the results of
+    multiple queries, while providing explicit logic for indexing.
+    """
+    def __init__(self, *args, **kwargs):
+        self.results = []
+        self.searches = {}
+        self.primary_search = None
+
+    def __getitem__(self, n):
+        if n < len(self.results):
+            return self.results[n]
+        elif n - len(self.results) > 1:
+            for i in range(len(self.results), n):
+                self[i]
```

Comment:
A comment like that would help out the next poor programmer who stumbles upon this oasis of elegance.